test/ui/ThumbnailTest.ts

```diff
@@ -67,22 +69,31 @@ export function ThumbnailTest() {
       });
     });
 
-    it('should set a CSS class when no thumbnail is available', () => {
-      var result = <IQueryResult>{
-        flags: ''
+    it('should instanciate an icon when no thumnail is available', (done) => {
+      let result = <IQueryResult>{ flags: '' };
+      let IconModuleMock: any = function () {
+        expect(true).toBe(true);
+        done();
       };
-      test = Mock.optionsResultComponentSetup<Thumbnail, IThumbnailOptions>(Thumbnail, undefined, result);
-      expect($$(test.cmp.img).hasClass('coveo-no-thumbnail')).toBe(true);
+
+      let envBuilder = new MockEnvironmentBuilder().withResult(result);      
+      new Thumbnail(test.env.element, {}, test.cmp.bindings, envBuilder.result, IconModuleMock);
     });
 
     describe('exposes options', () => {
-      it('noThumbnailClass should set the appropriate CSS class when no thumbnail is available', () => {
-        test = Mock.optionsResultComponentSetup<Thumbnail, IThumbnailOptions>(Thumbnail, <IThumbnailOptions>{
-          noThumbnailClass: 'coveo-heyo-there-is-no-class'
-        }, <IQueryResult>{ flags: '' });
+      it('noThumbnailClass should set the appropriate CSS class when no thumbnail is available', (done) => {
+        endpoint.getRawDataStream = () => new Promise<ArrayBuffer>((resolve, reject) => { reject(); });
+
+        test = Mock.advancedResultComponentSetup<Thumbnail>(Thumbnail, undefined, <Mock.AdvancedComponentSetupOptions>{
```

Comment:
Is this needed ? 

Can you do `test.env.endpoint.getRawDataStream = () => new Promise<ArrayBuffer>((resolve, reject) => { reject(); });` ?